external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/DirectKafkaInputDStream.scala

```diff
@@ -130,8 +130,15 @@ private[spark] class DirectKafkaInputDStream[K, V](
     offsets: Map[TopicPartition, Long]): Option[Map[TopicPartition, Long]] = {
     val estimatedRateLimit = rateController.map(_.getLatestRate())
 
+    // TODO: Fix in rate controller (need SparkConf or initial rate member value)
+    // Fix: Rate controller first value is -1, if -1 return use the default rate
+    val fixedEstimatedRateLimit = estimatedRateLimit match {
```

Comment:
Not so much of a hack, but limit reusability. The best place would be directly in the RateEstimator, but we don't have access to the "spark.streaming.backpressure.initialRate".